packages/tasks_panel/test/misc/e2e-config.spec.ts

```diff
@@ -0,0 +1,472 @@
+      mockUpdateConfiguration.expects("updateTasksConfiguration").withArgs(wsFolder).resolves();
+    });
+
+    afterEach(() => {
+      mockUpdateConfiguration.verify();
+    });
+
+    const tasks = [
+      {
+        label: "task1",
+        type: "type1",
+        __wsFolder: wsFolder,
+      },
+      {
+        label: "task2",
+        type: "type2",
+        __wsFolder: wsFolder,
+      },
+    ];
+    it("addTaskDefinition - get `tasks` returns `undefined`", async () => {
+      mockWorkspace
+        .expects("getConfiguration")
+        .withExactArgs("tasks", testVscode.Uri.file(wsFolder))
+        .returns(undefined);
+      await addTaskDefinition(wsFolder, tasks);
+      const _tasks = (<any>mockUpdateConfiguration).expectations["updateTasksConfiguration"][0].args[0][1];
+      expect(_tasks).be.deep.equal(tasks);
+    });
+
+    it("addTaskDefinition - update the existing tasks list", async () => {
+      const exisTasks = [{ label: "task3", type: "other" }];
+      mockWorkspace
+        .expects("getConfiguration")
+        .withExactArgs("tasks", testVscode.Uri.file(wsFolder))
+        .returns({ get: () => exisTasks });
+      await addTaskDefinition(wsFolder, tasks);
+      const _tasks = (<any>mockUpdateConfiguration).expectations["updateTasksConfiguration"][0].args[0][1];
+      expect(_tasks).be.deep.equal(concat(exisTasks, tasks));
+    });
+  });
+
+  describe("isFileExist scope", () => {
+    const url = testVscode.Uri.file("/home/user/test/project/my/file.txt");
+
+    let mockWorkspaceFs: SinonMock;
+    beforeEach(() => {
+      mockWorkspaceFs = sandbox.mock(testVscode.workspace.fs);
+    });
+
+    afterEach(() => {
+      mockWorkspaceFs.verify();
+    });
+
+    it("isFileExist - does exist", async () => {
+      mockWorkspaceFs.expects("stat").withExactArgs(url).resolves({ type: testVscode.FileType.File });
+      expect(await isFileExist(url)).be.true;
+    });
+
+    it("isFileExist - does NOT exist", async () => {
+      mockWorkspaceFs.expects("stat").withExactArgs(url).rejects(new Error("File not found"));
+      expect(await isFileExist(url)).be.false;
+    });
+  });
+
+  describe("generateMtaDeployTasks scope", () => {
+    const wsFolder = "/home/user/test/project/";
+    const project = "my";
+
+    let mockCfTools: SinonMock;
+    beforeEach(() => {
+      mockCfTools = sandbox.mock(cfTools);
+    });
+
+    afterEach(() => {
+      mockCfTools.verify();
+    });
+
+    const projectUri = testVscode.Uri.joinPath(testVscode.Uri.file(wsFolder), project);
+    const taskBuild = {
+      label: "Build MTA",
+      type: "build.mta",
+      taskType: "Build",
+      projectPath: `${projectUri.fsPath}`,
+      extensions: [],
+    };
+    const taskDeploy = {
+      type: "deploy.mta.cf",
+      label: `Deploy MTA to Cloud Foundry`,
+      taskType: "Deploy",
+      mtarPath: `${projectUri.fsPath}/mta_archives/${project || last(split(wsFolder, path.sep))}_0.0.1.mtar`,
```

Comment:
`project` is a constant string.
so why do we need the expression with `||`
`{project || last(split(wsFolder, path.sep))}` ?